panic for k8s scan

**Describe the bug**

spin up a new minikube and run the kubernetes security policy

**To Reproduce**

```
cnspec scan k8s -f core/mondoo-kubernetes-security.mql.yaml --output full --discover deploymentss
→ loaded configuration from /Users/chris/.config/mondoo/mondoo.yml using source default
! No credentials provided. Switching to --incognito mode.
→ discover related assets for 1 asset(s)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x40 pc=0x1031d9d10]

goroutine 1 [running]:
go.mondoo.com/cnspec/apps/cnspec/cmd.glob..func13(0x0?, 0x1037ccd90?, 0x104279be0?)
	/Users/chris/go/src/go.mondoo.com/cnspec/apps/cnspec/cmd/scan.go:159 +0x240
go.mondoo.com/cnquery/cli/providers.setConnector.func2(0x1400097a900?, {0x1400014b560?, 0x0, 0x6})
	/Users/chris/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20231002111758-82a0179e7dbd/cli/providers/providers.go:465 +0x854
github.com/spf13/cobra.(*Command).execute(0x14000971800, {0x1400014b500, 0x6, 0x6})
	/Users/chris/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944 +0x640
github.com/spf13/cobra.(*Command).ExecuteC(0x104149560)
	/Users/chris/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x320
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/chris/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
go.mondoo.com/cnspec/apps/cnspec/cmd.Execute()
	/Users/chris/go/src/go.mondoo.com/cnspec/apps/cnspec/cmd/root.go:109 +0x15c
main.main()
	/Users/chris/go/src/go.mondoo.com/cnspec/apps/cnspec/cnspec.go:9 +0x1c
```

**Expected behavior**

Scan runs as expected

--- a/apps/cnspec/cmd/scan.go
+++ b/apps/cnspec/cmd/scan.go
@@ -156,12 +156,14 @@ var scanCmdRun = func(cmd *cobra.Command, runtime *providers.Runtime, cliRes *pl
 
 	// if we had asset errors, we return a non-zero exit code
 	// asset errors are only connection issues
-	if len(report.Errors) > 0 {
-		os.Exit(1)
-	}
+	if report != nil {
+		if len(report.Errors) > 0 {
+			os.Exit(1)
+		}
 
-	if report.GetWorstScore() < uint32(conf.ScoreThreshold) {
-		os.Exit(1)
+		if report.GetWorstScore() < uint32(conf.ScoreThreshold) {
+			os.Exit(1)
+		}
 	}
 }
 

